### DISCUSSION TITLE
Disable cookie processing

### DISCUSSION THREAD

**Discussion by Kaspek2480** [aio-libs/aiohttp#8056] [Q&A]
Hi, i want to process cookies per requests

I use one ClientSession to use pooling

I have my own `Session` implementation and i handle cookies manually but the problem is some cookies are shared between requests

```
self.aiohttp_session = aiohttp.ClientSession(connector_owner=False, connector=TCPConnector(ssl=False))
```

Here's how i pass my own cookies to request:

```
response_obj = await self.aiohttp_session.request(
                url=url,
                method=self.request.method,
                headers=self.request.headers,
                params=query_params,

                data=self.request.body,
                cookies=cookies,
                proxy=proxy_obj,
                proxy_auth=proxy_auth_obj,
                ssl=False,
                timeout=my_timeout
            )
```
            
            How can i just disable aiohttp cookies processing/sharing?
            
            https://github.com/aio-libs/aiohttp/pull/963/commits/203449563e8734d0e7f20f04d0af7b3a2e4a3f85
            
            I found that but i'm not sure how can i use that

**Comment by Dreamsorcerer** [comment]
I think just subclass the regular cookie jar and make update_cookies() a no-op. From a brief glance at the code, I think that'll work.

**Reply by Dreamsorcerer** [reply]
Specifically, this is the code that loads your cookies onto the regular cookies:
https://github.com/aio-libs/aiohttp/blob/a379e6344432d5c033f78c2733fe69659e3cff50/aiohttp/client.py#L481-L488

And, this appears to be where the regular cookies get updated:
https://github.com/aio-libs/aiohttp/blob/a379e6344432d5c033f78c2733fe69659e3cff50/aiohttp/client.py#L575

**Comment by Kaspek2480** [comment]
Okay solved using that code

```py
class EmptyCookieJar(aiohttp.cookiejar.AbstractCookieJar):

    def __len__(self):
        pass

    def __iter__(self):
        pass

    def clear(self, predicate: Optional[ClearCookiePredicate] = None) -> None:
        pass

    def clear_domain(self, domain: str) -> None:
        pass

    def update_cookies(self, cookies: LooseCookies, response_url: URL = URL()) -> None:
        pass

    def filter_cookies(self, request_url: URL) -> "BaseCookie[str]":
        return BaseCookie()
```

and then

```py
aiohttp.ClientSession(connector_owner=False, cookie_jar=EmptyCookieJar())
```

**Comment by Kaspek2480** [comment]
I have another problem, when i used custom cookie jar, and i update cookies manually by fetching response, the thing is not working when it auto redirects - what can i do in that case?
